### ISSUE TITLE
When trying to use a placeholder node from `get_children` the node is already an instance

### ISSUE THREAD

**Issue by Tadaboody** [godotengine/godot#43209]
**Godot version:**
`3.2.3.stable.official`

**OS/device including version:**

OSX 10.15.7

**Issue description:**

When trying to use a placeholder node from `get_children` the type returned is not a `InstancePlaceholder` but the type of the instance

**Steps to reproduce:**
- Have a scene with a parent node and a child node
- Make the child node a placeholder
- run `$Parent.get_children()[0].create_instance()` in `_process`
- crash

**Minimal reproduction project:**

[child_placehoder_example.zip](https://github.com/godotengine/godot/files/5465575/child_placehoder_example.zip)

### TLDR code example:
```
extends Node2D


func _ready():
	$Container.get_children()[0].create_instance()  # This works 


func _process(delta):
	$Container/Scene.create_instance()  # This works
	$Container.get_children()[0].create_instance()  # Already a Node2D and not an InstancePlaceholder!
```

**Comment by Lielay9** [comment]
Came across this issue when searching about InstancePlaceholders. 
```gdscript
func _process(delta):
	...
	$Container.get_children()[0].create_instance()  # Already a Node2D and not an InstancePlaceholder!
```
The reason why the above doesn't work is that the InstancePlaceholder pushes the scene above itself. I.e. if the InstancePlaceholder was previously a child at index 0 it will be at index 1 after creating an instance.


```gdscript
func _process(delta):
	$Container/Scene.create_instance()  # This works
	...
```
The reason this works is that it references the InstancePlaceholder by its name (which it stole from the child scene root).

For future reference, the reference to InstancePlaceholder should be saved to a variable or replaced entirely if it's no longer necessary.

**TDLR:** This is not a bug and can be closed.
